File: backend/app/documents/partnership.py
"""Chat-driven field extraction for the Partnership Agreement.

Scope note: the real template also defines advanced/rarely-used terms
(Increased Claims, Increased Cap Amount, Unlimited Claims, Additional
Warranties, an optional DPA/Brand Guidelines reference) that this chat does
not ask about. Those substitute to fixed defaults in the frontend's
substitution map rather than becoming extra chat questions, matching the
depth already used for the Mutual NDA.
"""

import datetime

from pydantic import BaseModel

from app.chat import ChatMessage
from app.chat import run_chat_turn as _run_chat_turn
from app.documents.common import CHAT_RULES, PartyFields

SLUG = "partnership-agreement"
CATALOG_NAME = "Partnership Agreement"
TEMPLATE_FILE = "Partnership-Agreement.md"
DOCUMENT_TITLE = "Partnership Agreement"


class PartnershipChatFields(BaseModel):
    effectiveDate: str | None = None
    endDate: str | None = None
    governingLaw: str | None = None
    chosenCourts: str | None = None
    obligations: str | None = None
    paymentProcess: str | None = None
    paymentSchedule: str | None = None
    territory: str | None = None
    generalCapAmount: str | None = None
    company: PartyFields | None = None
    partner: PartyFields | None = None


class PartnershipChatTurn(BaseModel):
    reply: str
    fields: PartnershipChatFields
    done: bool


SYSTEM_PROMPT = f"""You are an assistant helping a user fill in a Partnership Agreement covering
mutual business obligations and trademark licensing between two companies.

Today's date is {datetime.date.today().isoformat()}.

Ask about the following fields, ONE QUESTION AT A TIME, in this fixed order:
1. effectiveDate - the date the agreement starts (always output as YYYY-MM-DD)
2. endDate - the date the agreement ends (YYYY-MM-DD)
3. governingLaw and chosenCourts
4. obligations - a short description of what each party must do
5. paymentProcess - how invoices/billing work, or "None" if no payments are involved
6. paymentSchedule - when payments are due, or "None" if no payments are involved
7. territory - the geographic territory the partnership covers
8. generalCapAmount - the total liability cap for either party (e.g. a dollar amount)
9. company: name, title, company, and noticeAddress - ask for these together as one question
10. partner: name, title, company, and noticeAddress - ask for these together as one question
{CHAT_RULES}"""


def run_chat_turn(messages: list[ChatMessage]) -> PartnershipChatTurn:
    return _run_chat_turn(SYSTEM_PROMPT, messages, PartnershipChatTurn)

File: frontend/lib/documents/partnership.ts
import type { DocumentModule } from "./types";
import {
  blankParty,
  formatLongDate,
  getParty,
  mergeSubstitutions,
  roleAliases,
  standardLiabilityDefaults,
  textValue,
} from "./common";

export const partnership: DocumentModule = {
  slug: "partnership-agreement",
  catalogName: "Partnership Agreement",
  templateFile: "Partnership-Agreement.md",
  documentTitle: "Partnership Agreement",
  subtitle: "Common Paper Partnership Agreement Standard Terms Version 1.0",
  defaultValues: {
    effectiveDate: "",
    endDate: "",
    governingLaw: "",
    chosenCourts: "",
    obligations: "",
    paymentProcess: "None.",
    paymentSchedule: "None.",
    territory: "",
    generalCapAmount: "",
    company: { ...blankParty },
    partner: { ...blankParty },
  },
  groups: [
    {
      title: "Cover Page",
      fields: [
        { label: "Effective Date", render: (v) => formatLongDate(v.effectiveDate) },
        { label: "End Date", render: (v) => formatLongDate(v.endDate) },
        { label: "Governing Law", key: "governingLaw" },
        { label: "Chosen Courts", key: "chosenCourts" },
        { label: "Obligations", key: "obligations" },
        { label: "Payment Process", key: "paymentProcess" },
        { label: "Payment Schedule", key: "paymentSchedule" },
        { label: "Territory", key: "territory" },
        { label: "General Cap Amount", key: "generalCapAmount" },
        { label: "Company", key: "company", party: true },
        { label: "Partner", key: "partner", party: true },
      ],
    },
  ],
  buildSubstitutions: (values) => {
    const company = getParty(values, "company");
    const partner = getParty(values, "partner");
    return mergeSubstitutions(
      {
        keyterms_link: {
          "Effective Date": formatLongDate(values.effectiveDate),
          "Governing Law": textValue(values, "governingLaw", "[Governing Law]"),
          "Chosen Courts": textValue(values, "chosenCourts", "[Chosen Courts]"),
          "General Cap Amount": textValue(values, "generalCapAmount", "[General Cap Amount]"),
        },
        businessterms_link: {
          Obligations: textValue(values, "obligations", "[Obligations]"),
          "Payment Process": textValue(values, "paymentProcess", "None."),
          "Payment Schedule": textValue(values, "paymentSchedule", "None."),
          Territory: textValue(values, "territory", "[Territory]"),
          "End Date": formatLongDate(values.endDate),
        },
      },
      { keyterms_link: standardLiabilityDefaults("Company", "Partner") },
      { keyterms_link: { "Brand Guidelines": "Company's standard brand guidelines" } },
      { keyterms_link: roleAliases("Company", company) },
      { keyterms_link: roleAliases("Partner", partner) },
    );
  },
};
